homeassistant/components/mqtt/client.py

```diff
@@ -893,14 +893,35 @@ async def _async_perform_subscriptions(self) -> None:
         if not self._pending_subscriptions:
             return
 
-        subscriptions: dict[str, int] = self._pending_subscriptions
+        pending_subscriptions: dict[str, int] = self._pending_subscriptions
+        pending_wildcard_subscriptions = {
+            subscription.topic: pending_subscriptions.pop(subscription.topic)
+            for subscription in self._wildcard_subscriptions
+            if subscription.topic in pending_subscriptions
+        }
+        simple_subscription_list = list(pending_subscriptions.items())
+
         self._pending_subscriptions = {}
 
-        subscription_list = list(subscriptions.items())
         debug_enabled = _LOGGER.isEnabledFor(logging.DEBUG)
 
-        for chunk in chunked_or_all(subscription_list, MAX_SUBSCRIBES_PER_CALL):
+        for topic, qos in pending_wildcard_subscriptions.items():
```

Comment:
Instead of adding another for loop, can we use `itertools.chain` in the existing for loop below to iterate both over the wildcard subscriptions and the chunked normal subscriptions?

Reply:
That is not as simple as it looks as we need a list conversion on the chunk, but on a tuple, `list` will unpack the tuple into a list.